packages/explorer/src/app/worlds/[worldAddress]/interact/Form.tsx

```diff
@@ -0,0 +1,85 @@
+"use client";
+
+import { Coins, Eye, Send } from "lucide-react";
+import { AbiFunction } from "viem";
+import { useDeferredValue, useState } from "react";
+import { Input } from "../../../../components/ui/Input";
+import { useHashState } from "../../../../hooks/useHashState";
+import { cn } from "../../../../lib/utils";
+import { FunctionField } from "./FunctionField";
+
+type Props = {
+  data: {
+    abi: AbiFunction[];
```

Comment:
ooc why is this nested inside `data`?

Reply:
Generally, I try to pass only the least required props and when it comes to backend response, prefer to destructure it in the component where it's used. But tbh, probably applying the principle of "least-privilege" makes sense even here. Will apply it moving forward